Uyanda.Coffee.Persistence/Accessors/BeverageAccessor.cs

```diff
@@ -101,8 +101,8 @@ public async Task<IEnumerable<LineItemModel>> PurchaseAsync(IEnumerable<LineItem
             localDbContext.Invoice.Add(invoice);
 
             await localDbContext.SaveChangesAsync();
-           
-            return purchase.Select(ToModel).ToArray();
+            IEnumerable<InvoiceEntity> result = new[] { invoice };
+            return result.Select(ToModel).First();
```

Comment:
Uyanda . . . Let us think of what is happening here . . . You are creating an Array, only to take the first element of it.
Rather you just have _return ToModel(invoice);_. The in between result Array is not necessary.

Reply:
Noted.